test/acceptance/features/steps/kubernetesapi.py

```diff
@@ -0,0 +1,89 @@
+import time
+from behave import then
+from datetime import datetime, timedelta
+from kubernetes import client
+
+
+@then(u'On Primaza Cluster "{cluster}", ClusterEnvironment "{ce_name}" state will eventually move to "{state}"')
+@then(u'On Primaza Cluster "{cluster}", ClusterEnvironment "{ce_name}" state will move to "{state}" in "{timeout}" seconds')
+def on_primaza_cluster_check_state(context, cluster, ce_name, state, timeout=60):
+    api_client = context.cluster_provider.get_primaza_cluster(cluster).get_api_client()
+    cobj = client.CustomObjectsApi(api_client)
+
+    tend = datetime.now() + timedelta(seconds=timeout)
+    while datetime.now() < tend:
+        ce_status = cobj.get_namespaced_custom_object_status(
+            group="primaza.io",
+            version="v1alpha1",
+            namespace="primaza-system",
+            plural="clusterenvironments",
+            name=ce_name)
+        ce_state = ce_status.get("status", {}).get("state", None)
+        print(f"ClusterEnvironment '{ce_name}' status is '{ce_state}', wanted '{state}'")
+        if ce_state is not None and ce_state == state:
+            return
+        time.sleep(5)
```

Comment:
This can be rewritten to use `polling2.poll`; if you want to punt that to a future PR, that's fine with me.

Reply:
nice suggestion, I'm going to use polling2 in this PR